corehq/apps/app_manager/tests/test_form_preparation_v2.py

```diff
@@ -122,25 +123,39 @@ def test_subcase_multiple_repeats(self):
                             self.get_xml('multiple_subcase_repeat'))
 
     def test_subcase_repeat_mixed_form(self):
-        self.app = Application.wrap(self.get_json('subcase_repeat_mixed'))
-        self.assertXmlEqual(self.get_xml('subcase_repeat_mixed_form'),
-                            self.app.get_module(0).get_form(0).render_xform())
-
-    def test_subcase_repeat_mixed_suite(self):
-        # should this be in the suite test?
-        self.app = Application.wrap(self.get_json('subcase_repeat_mixed'))
-        expected = """
-        <partial>
-            <session>
-              <datum id="case_id_new_parent_0" function="uuid()"/>
-              <datum id="case_id_new_subcase_1" function="uuid()"/>
-              <datum id="case_id_new_subcase_3" function="uuid()"/>
-            </session>
-        </partial>
-        """
-        self.assertXmlPartialEqual(expected,
-                                   self.app.create_suite(),
-                                   './entry[1]/session')
+        app = Application.new_app(None, "Untitled Application", application_version=APP_V2)
+        module_0 = app.add_module(Module.new_module('parent', None))
+        module_0.unique_id = 'm0'
+        module_0.case_type = 'parent'
+        form = app.new_form(0, "Form", None, attachment=self.get_xml('subcase_repeat_mixed_form_pre'))
+
+        module_1 = app.add_module(Module.new_module('subcase', None))
+        module_1.unique_id = 'm1'
+        module_1.case_type = 'subcase'
+
+        form.actions.open_case = OpenCaseAction(name_path="/data/parent_name")
+        form.actions.open_case.condition.type = 'always'
+
+        form.actions.subcases.append(OpenSubCaseAction(
+            case_type=module_1.case_type,
+            case_name="/data/first_child_name",
+            condition=FormActionCondition(type='always')
+        ))
+        # subcase in the middle that has a repeat context
+        form.actions.subcases.append(OpenSubCaseAction(
+            case_type=module_1.case_type,
+            case_name="/data/repeat_child/repeat_child_name",
+            repeat_context='/data/repeat_child',
+            condition=FormActionCondition(type='always')
+        ))
+        form.actions.subcases.append(OpenSubCaseAction(
+            case_type=module_1.case_type,
+            case_name="/data/last_child_name",
+            condition=FormActionCondition(type='always')
+        ))
```

Comment:
Nice, I like this style of creating apps in code, in the test. It makes it much clearer what you're testing.